joern-cli/frontends/jimple2cpg/src/test/scala/io/joern/jimple2cpg/unpacking/JarUnpacking.scala

```diff
@@ -1,57 +1,70 @@
 package io.joern.jimple2cpg.unpacking
 
+import better.files.File
 import io.joern.jimple2cpg.{Config, Jimple2Cpg}
 import io.joern.jimple2cpg.util.ProgramHandlingUtil
 import io.shiftleft.codepropertygraph.Cpg
-import io.shiftleft.semanticcpg.language._
+import io.shiftleft.semanticcpg.language.*
 import io.shiftleft.utils.ProjectRoot
 import org.scalatest.BeforeAndAfterAll
 import org.scalatest.matchers.must.Matchers
-import org.scalatest.matchers.should.Matchers._
+import org.scalatest.matchers.should.Matchers.*
 import org.scalatest.wordspec.AnyWordSpec
 
 import java.nio.file.{Files, Path, Paths}
 import scala.util.{Failure, Success, Try}
 
 class JarUnpacking extends AnyWordSpec with Matchers with BeforeAndAfterAll {
 
-  var cpg: Cpg = _
+  var validCpgs: Map[String, Cpg] = _
+  var slippyCpg: Cpg              = _
 
   override protected def beforeAll(): Unit = {
     super.beforeAll()
-    Try(getClass.getResource("/unpacking").toURI) match {
+    validCpgs = List("HelloWorld.jar", "NestedHelloWorld.jar", "helloworld")
+      .map(k => (k, getUnpackingCpg(k)))
+      .toMap
+    slippyCpg = getUnpackingCpg("slippy.zip")
+  }
+
+  private def getUnpackingCpg(path: String): Cpg =
+    Try(getClass.getResource(s"/unpacking/${path}").toURI) match {
       case Success(x) =>
         implicit val defaultConfig: Config = Config()
-        cpg = new Jimple2Cpg().createCpg(Paths.get(x).toString).get
+        new Jimple2Cpg().createCpg(Paths.get(x).toString).get
       case Failure(x: Throwable) =>
         fail("Unable to obtain test resources.", x)
     }
-  }
 
-  "should extract files and clean up temp directory" in {
+  "'resources/unpacking/helloworld' should contain 'HelloWorld.jar'" in {
     val targetDir = ProjectRoot.relativise("joern-cli/frontends/jimple2cpg/src/test/resources/unpacking")
-    Files
-      .walk(Path.of(targetDir))
-      .map(_.toFile)
-      .filter(f => f.isFile && f.getName.contains(".jar"))
-      .map(_.getName)
-      .toArray shouldBe Array("HelloWorld.jar")
-    ProgramHandlingUtil.getUnpackingDir.toFile.length() shouldBe 0
+    File(targetDir)
+      .walk()
+      .filter(f => f.isRegularFile && f.extension.exists(_ == ".jar"))
+      .map(_.name)
+      .toSet shouldBe Set("HelloWorld.jar", "NestedHelloWorld.jar")
   }
 
   "should reflect the correct package order" in {
-    val List(foo) = cpg.typeDecl.fullNameExact("Foo").l
-    foo.name shouldBe "Foo"
-
-    val List(bar) = cpg.typeDecl.fullNameExact("pac.Bar").l
-    bar.name shouldBe "Bar"
-
-    cpg.method.filterNot(_.isExternal).fullName.toSetMutable shouldBe Set(
-      "Foo.<init>:void()",
-      "Foo.add:int(int,int)",
-      "pac.Bar.sub:int(int,int)",
-      "pac.Bar.<init>:void()"
-    )
+    for ((name, cpg) <- validCpgs) {
```

Comment:
Wasn't sure how the include `name` in ScalaTest.

Reply:
That is fine, this kind of info isn't really embedded in the CPG. In source code frontends, you'd get the project root under `cpg.metaData.root`

Reply:
I meant more about having test failure attribution. I don't know if there is such a way to define sub tests or something like that